source/slang/slang-ir-validate.cpp

```diff
@@ -223,6 +223,13 @@ namespace Slang
         {
         case kIROp_DifferentiableTypeDictionaryItem:
             return;
+        default:
+            // Variables which eventually resolve into global params will always
+            // have an ancestor referenced at some point, these global params may
+            // not be resolved yet, do not exception on these types.
+            if (inst->getOperand(0)->findDecoration<IRLayoutDecoration>())
+                return;
+            break;
```

Comment:
I don't understand what this comment means.
Please make it easier to read and clear for the meaning.